src/videojs-hls.js

```diff
@@ -350,15 +340,21 @@ videojs.Hls.prototype.setupSourceBuffer_ = function() {
 
     timelineUpdates.forEach(function (update) {
       if (segment) {
-        if (update.start !== undefined) {
-          segment.start = update.start;
-        }
         if (update.end !== undefined) {
           segment.end = update.end;
         }
       }
     });
 
+    // if we've buffered to the end of the video, let the MediaSource know
+    if (this.playlists.media().endList &&
+        currentBuffered.length &&
+        segments[segments.length - 1].end <= currentBuffered.end(0) &&
+        this.mediaSource.readyState === 'open') {
+      this.mediaSource.endOfStream();
```

Comment:
Basically we say: If we aren't in a live playlist and the last segment has a duration equal to the end point of the currently buffered region, we consider the video to have ended.